packages/contracts/contracts/plugin/IPluginSetup.sol

```diff
@@ -31,46 +44,28 @@ interface IPluginSetup {
+    /// @param _payload TOD:GIORGI how to add explanation ?
     /// @return initData The initialization data to be passed to upgradeable contracts when the update is applied in the `PluginSetupProcessor`.
-    /// @return permissions The array of multi-targeted permission operations to be applied by the `PluginSetupProcessor` to the updating DAO.
-    /// @dev The array of `_currentHelpers` has to be specified in the same order as they were returned from previous setups preparation steps (the latest `prepareInstallation` or `prepareUpdate` step that has happend) on which this update is prepared for.
+    /// @return preparedDependency TOD:GIORGI
     function prepareUpdate(
         address _dao,
-        address _plugin,
-        address[] memory _currentHelpers,
-        uint16[3] calldata _oldVersion,
-        bytes memory _data
-    )
-        external
-    
-        returns (
-            address[] memory updatedHelpers,
-            bytes memory initData,
-            PermissionLib.ItemMultiTarget[] memory permissions
-        );
+        uint16 _currentBuild,
+        SetupPayload calldata _payload
+    ) external returns (bytes memory initData, PreparedDependency memory preparedDependency);
 
     /// @notice The ABI required to decode the `bytes` data in `prepareUninstallation()`.
     /// @return The ABI in string format.
     function prepareUninstallationDataABI() external view returns (string memory);
 
     /// @notice Prepares the uninstallation of a plugin.
     /// @param _dao The address of the uninstalling DAO.
-    /// @param _plugin The address of the `Plugin` contract to update from.
-    /// @param _currentHelpers The address array of all current helpers (contracts or EOAs) associated with the plugin to update from.
-    /// @param _data The `bytes` encoded data containing the input parameters for the uninstalltion as specified in the `prepareUninstallationDataABI()` function.
+    /// @param _payload TOD:GIORGI how to add explanation ?
     /// @return permissions The array of multi-targeted permission operations to be applied by the `PluginSetupProcessor` to the uninstalling DAO.
-    /// @dev The array of `_currentHelpers` has to be specified in the same order as they were returned from previous setups preparation steps (the latest `prepareInstallation` or `prepareUpdate` step that has happend) on which this update is prepared for.
-    function prepareUninstallation(
-        address _dao,
-        address _plugin,
-        address[] calldata _currentHelpers,
-        bytes calldata _data
-    ) external returns (PermissionLib.ItemMultiTarget[] memory permissions);
+    /// @param actions The array of follow up actions that will be executed through dao once the plugin is installed/updated/uninstalled.
+    function prepareUninstallation(address _dao, SetupPayload calldata _payload)
+        external
+        returns (PermissionLib.ItemMultiTarget[] memory permissions, IDAO.Action[] memory actions);
```

Comment:
Shouldn't we still return `PreparedDependency` even if helpers will always be empty?
The name "PreparedDependency" would need to be changed then.

Reply:
What other idea do you have ? Note that `prepareDependency` contains `helpers`, `permissions`, `actions`. If I return it then it means it returns helpers as well, though from `prepareUninstallation` , we don't need helpers. 

Happy to hear your opinions.